src/components/Exchange/SwapBox.js

```diff
@@ -1642,10 +1643,14 @@ export default function SwapBox(props) {
   };
 
   const onSwapOptionChange = (opt) => {
+    // Keep same fromToken when toggling swap option
+    const _swapOption = swapOption;
+    const prevTokenAddress = tokenSelection[_swapOption].from;
+    const updatedTokenSelection = JSON.parse(JSON.stringify(tokenSelection));
+    updatedTokenSelection[opt].from = prevTokenAddress;
+    setTokenSelection(updatedTokenSelection);
     setSwapOption(opt);
     setAnchorOnFromAmount(true);
-    setFromValue("");
-    setToValue("");
```

Comment:
surely these have to remain the same if there arent any other changes?

Reply:
These two lines clear the value - Kurt requested that the value be retained across swap options, so by removing these the value stays the same